### PR TITLE
Add invert function

### PR DESCRIPTION
**Pull request by agao48**
**Describe your changes**
Add `invert`

**Testing performed**
Tests added

### CODE REVIEW

File: expr/func_invert.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+func (s *FuncInvert) Exec(dataMap DataMap) ([]models.Series, error) {
+	series, err := s.in.Exec(dataMap)
+	if err != nil {
+		return nil, err
+	}
+
+	for i, serie := range series {
+		series[i].Target = fmt.Sprintf("invert(%s)", serie.Target)
```
Comment (shanson7) [inline expr/func_invert.go]: We can't modify these in place. as it changes the input. Can you add a test that verifies the input is unmodified after running through this function?

**Review by shanson7 — APPROVED** [review]

File: expr/func_invert_test.go [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+		{
+			Interval:   10,
+			QueryPatt:  "queryPattHere",
+			Target:     "targetHere",
+			Datapoints: getCopy(datapoints),
+		},
+	}
+
+	// store copy of the original input
+	inCopy := []models.Series{
```
Comment (shanson7) [inline expr/func_invert_test.go]: Did you verify that this failed before the fix?

Comment (agao48) [reply expr/func_invert_test.go]: Yes, all the fields were getting modified in-place.

**Comment by shanson7** [comment]
Merged upstream 1791

**Pull request closed without merging**